### ISSUE TITLE
Potential leak in auth plugins

### ISSUE THREAD

**Issue by alanking** [irods/irods#6375]
- [ ] main
- [ ] 4-2-stable
---
The `auth_scheme` is copied into the comm object for the agent request step of the authentication flow as far back as 4.2.0:
https://github.com/irods/irods/blob/f9c8c99c1188f92bc109da46f1eb2ca9e9ad9e04/plugins/auth/native/libnative.cpp#L297
https://github.com/irods/irods/blob/f9c8c99c1188f92bc109da46f1eb2ca9e9ad9e04/plugins/auth/pam/libpam.cpp#L378
https://github.com/irods/irods/blob/f9c8c99c1188f92bc109da46f1eb2ca9e9ad9e04/plugins/auth/osauth/libosauth.cpp#L326
https://github.com/irods/irods_auth_plugin_kerberos/blob/db961bf75615d8eefd497fe840d7494e106e2126/krb/libkrb.cpp#L1181

Please investigate whether this results in a leak.

**Comment by alanking** [comment]
I think I found the `free` call, so this might be close-able...

https://github.com/irods/irods/blob/e874106bbb1b67d7b59e7cbd112d37d200c2710a/server/core/src/rodsAgent.cpp#L593

**Comment by korydraughn** [comment]
Yes. That call to `free()` guards against a memory leak if everything works as expected.

Updating the code to use `at_scope_exit` would be better, but I don't think we need to address this for 4.2.12.

Bumping to 4.3.2.

**Comment by trel** [comment]
updated the checkboxes accordingly.
